src/backend/internal/third-party/s3utils/s3utils.go

```diff
@@ -0,0 +1,127 @@
+package s3utils
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"sync"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+)
+
+type BucketBasics struct {
+	S3Client *s3.Client
+}
+
+const awsRegion = "us-east-1"
+
+var requiredBuckets = []string{
+	"profile-picture",
+}
+
+var (
+	instance *BucketBasics
+	once     sync.Once
+)
+
+func createS3Client() (*s3.Client, error) {
+	awsEndpoint := "http://localhost:4566"
+
+	customResolver := aws.EndpointResolverWithOptionsFunc(func(service, region string, options ...interface{}) (aws.Endpoint, error) {
+		if awsEndpoint != "" {
+			return aws.Endpoint{
+				PartitionID:   "aws",
+				URL:           awsEndpoint,
+				SigningRegion: awsRegion,
+			}, nil
+		}
+
+		// returning EndpointNotFoundError will allow the service to fallback to its default resolution
+		return aws.Endpoint{}, &aws.EndpointNotFoundError{}
+	})
+
+	awsCfg, err := config.LoadDefaultConfig(context.TODO(),
```

Comment:
The purpose of `context.TODO()` is for when you haven't decided yet which context you should use in this function, in this case please remain it as it is.

On the contrary, if you meant to use it as a root context, for when no other context is available, please use `context.Background()`.